app/src/main/java/waiter/App.java

```diff
@@ -3,12 +3,16 @@
  */
 package waiter;
 
+import java.io.IOException;
+import java.net.ServerSocket;
+
 public class App {
-    public String getGreeting() {
-        return "Hello World!";
-    }
 
-    public static void main(String[] args) {
-        System.out.println(new App().getGreeting());
+    public static void main(String[] args) throws IOException {
+        int port = 4424;
+        Listener listener = new Listener(new ServerSocket(port));
+        Messenger messenger = new Messenger(new EchoProtocol());
+        Communicator communicator = new Communicator();
+        communicator.communicate(listener, messenger);
```

Comment:
What happens after the echo message is sent? does the server terminates the connection and quit?

Reply:
The server stops running and the program exits

Reply:
Make sense, I was thinking if we need to do some clean up like closing the connection but it looks like we're ok right now since the program terminates itself.

Reply:
Just to keep track of this as well, this will be addressed in the Multiple Message ticket